File: JDK11/opentraceability/src/main/java/opentraceability/models/masterdata/kdes/MasterDataKDEObject.java
package opentraceability.models.masterdata.kdes;

import opentraceability.interfaces.IMasterDataKDE;
import opentraceability.utility.XmlToJsonConverter;
import org.json.JSONObject;
import org.w3c.dom.Element;
import java.lang.reflect.Type;

public class MasterDataKDEObject extends IMasterDataKDE  {
    private Element _xml = null;
    private JSONObject _json = null;

    Type valueType = Object.class;

    public Object getValue() {
        return _xml != null ? _xml : _json;
    }

    public void setValue(Object value) {
        if (value instanceof Element) {
            _xml = (Element) value;
            _json = null;
        } else if (value instanceof JSONObject) {
            _xml = null;
            _json = (JSONObject) value;
        } else {
            throw new IllegalArgumentException("Unsupported value type: " + value.getClass());
        }
    }

    public MasterDataKDEObject() {}

    public MasterDataKDEObject(String ns, String name) {
        super.namespace = ns;
        super.name = name;
    }

    public void setFromGS1WebVocabJson(JSONObject json) {
        _xml = null;
        _json = json;
    }

    public JSONObject getGS1WebVocabJson() {
        if (_xml != null) {
            JSONObject json = XmlToJsonConverter.toJSON(_xml);
            return json;
        } else {
            return _json;
        }
    }

    public void setFromEPCISXml(Element xml) {
        _xml = xml;
        _json = null;
    }

    public Element getEPCISXml() {
        Element xml;
        if (_xml != null) {
            xml = _xml;
        } else if (_json != null) {
            xml = XmlToJsonConverter.toXML(_json);
        } else {
            xml = null;
        }
        
        return xml;
    }
}

File: CSharp/OpenTraceability/Models/MasterData/KDEs/MasterDataKDEObject.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using OpenTraceability.Interfaces;
using OpenTraceability.Models.MasterData.KDEs;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml;
using System.Xml.Linq;

namespace OpenTraceability.Models.Events.KDEs
{
    public class MasterDataKDEObject : MasterDataKDEBase, IMasterDataKDE
    {
        private XElement _xml = null;
        private JToken _json = null;

        public Type ValueType => typeof(object);
        public object Value
        {
            get
            {
                if (_xml != null)
                {
                    return _xml;
                }
                else
                {
                    return _json;
                }
            }
        }

        internal MasterDataKDEObject()
        {

        }

        public MasterDataKDEObject(string ns, string name)
        {
            this.Namespace = ns;
            this.Name = name;
        }

        public void SetFromGS1WebVocabJson(JToken json)
        {
            _xml = null;
            _json = json;
        }

        public JToken GetGS1WebVocabJson()
        {
            if (_xml != null)
            {
                // convert _xml to JObject
                var j = Newtonsoft.Json.JsonConvert.SerializeXNode(_xml);
                return j;
            }
            else if (_json != null)
            {
                return _json;
            }
            else
            {
                return null;
            }
        }

        public void SetFromEPCISXml(XElement xml)
        {
            _xml = xml;
            _json = null;
        }

        public XElement GetEPCISXml()
        {
            if (_xml != null)
            {
                return _xml;
            }
            else if (_json != null)
            {
                // convert _json to XElement
                string xmlStr = (JsonConvert.DeserializeXmlNode(_json.ToString()) as XmlDocument)?.OuterXml;
                if (!string.IsNullOrEmpty(xmlStr))
                {
                    return XElement.Parse(xmlStr);
                }
                else
                {
                    return null;
                }
            }
            else
            {
                return null;
            }
        }
    }
}
